source/common/redis/codec_impl.cc

```diff
@@ -232,11 +228,16 @@ void DecoderImpl::parseSlice(const Buffer::RawSlice& slice) {
           state_ = State::ValueStart;
         }
       } else if (current_value.value_->type() == RespType::Integer) {
-        current_value.value_->asInteger() = pending_integer_.integer_;
+        if (pending_integer_.integer_ == 0 || !pending_integer_.negative_) {
+          current_value.value_->asInteger() = pending_integer_.integer_;
+        } else {
+          current_value.value_->asInteger() =
```

Comment:
It's not immediately obvious what the math is doing to avoid overflow. Same for the encode case. Can you add some comments?